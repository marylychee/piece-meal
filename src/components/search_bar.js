import React, { Component } from 'react';
import { Navbar } from 'react-bootstrap';
import { FormGroup } from 'react-bootstrap';
import { FormControl } from 'react-bootstrap';
import { Button } from 'react-bootstrap';


class SearchBar extends Component {
    constructor(props) {
      super(props);

      this.state = { term: '' };
    }

    render() {
      return (
        <Navbar.Form pullLeft>
          <FormGroup>
            <FormControl type="text" placeholder="Search Recipes" />
          </FormGroup>
          {' '}
          <Button type="submit">Go</Button>
        </Navbar.Form>
      )
    }

    // onInputChange(term) {
    //   this.setState({term});
    //   this.props.onSearchTermChange(term)
    // }
}

export default SearchBar;
