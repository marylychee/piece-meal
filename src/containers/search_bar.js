import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setQuery, doSearch } from '../actions/index'
import { bindActionCreators } from 'redux';
import { Navbar, FormGroup, FormControl, Button } from 'react-bootstrap';

const mapStateToProps = (state) => {
  console.log('this is the state.query', state.query);
  return {
    value: state.query
  }
}

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({ setQuery, doSearch }, dispatch)
}

class SearchBar extends Component {
    render() {
      return (
        <Navbar.Form pullLeft onSubmit={(event) => {
          event.preventDefault();
          this.props.doSearch(this.props.value);
        }}>
          <FormGroup>
            <FormControl value={this.props.value} onChange={(event)=>this.props.setQuery(event.target.value)} type="text" placeholder="Search Recipes" />
          </FormGroup>
          {' '}
          <Button type="submit">Go</Button>
        </Navbar.Form>
      )
    }

    // onInputChange(term) {
    //   this.setState({term});
    //   this.props.onSearchTermChange(term)
    // }
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchBar);
