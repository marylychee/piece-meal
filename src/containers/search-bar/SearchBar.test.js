import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { SearchBar } from './search_bar';

describe('Search', () => {
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchBar>Search</SearchBar>, div);
  });

  test('snapshots', () => {
    const component = shallow(
      <SearchBar>Search</SearchBar>
    );
    expect(toJson(component)).toMatchSnapshot();

  });
});
