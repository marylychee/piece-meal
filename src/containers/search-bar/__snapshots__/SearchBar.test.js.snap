exports[`Search snapshots 1`] = `
<NavbarForm
  componentClass="div"
  pullLeft={true}
  pullRight={false}>
  <FormGroup
    bsClass="form-group">
    <FormControl
      bsClass="form-control"
      componentClass="input"
      onChange={[Function]}
      placeholder="Search Recipes"
      type="text" />
  </FormGroup>
   
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}>
    Go
  </Button>
</NavbarForm>
`;
